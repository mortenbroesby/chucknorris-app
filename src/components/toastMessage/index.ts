import Vue from "vue";
import { Component, Prop } from "vue-property-decorator";

import template from "./toastMessage.vue";
import "./toastMessage.scss";

@Component({
  mixins: [template],
  components: {}
})
export default class ToastMessage extends Vue {
  /*************************************************/
  /* EXTERNAL PROPERTIES */
  /*************************************************/
  @Prop({ default: "" })
  message: string;

  @Prop({ default: false })
  isVisible: boolean;
}
