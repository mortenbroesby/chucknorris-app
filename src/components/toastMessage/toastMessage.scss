@import "../../styles/global.scss";

$max-width: 440px;

.toastMessage {
  position: absolute;
  bottom: -150px;
  left: 0;

  .toastMessage--inner {
    position: relative;
    width: $max-width;
    max-width: $max-width;
    height: 100px;
    display: inline-block;
  }

  .toastItem {
    position: absolute;
    top: 0;
    left: 0;
    width: $max-width;
    max-width: $max-width;
    padding: 10px;
    color: color('dark-grey');

    &--title {
      font-size: 16px;
      font-weight: 600;
    }

    &--message {
      line-height: 1.5;
    }
  }
}
