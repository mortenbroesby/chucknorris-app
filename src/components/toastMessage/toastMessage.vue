<template>
  <div class="toastMessage">
    <div class="toastMessage--inner">
      <div class="toastItem" v-show="isVisible">
        <p class="toastItem--title">Note:</p>
        <p class="toastItem--message">{{ message }}</p>
      </div>
    </div>
  </div>
</template>
