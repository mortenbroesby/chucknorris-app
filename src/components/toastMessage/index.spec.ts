import { mount, createLocalVue } from "@vue/test-utils";

import toastMessage from "./index";

describe("components/toastMessage", () => {
  const localVue = createLocalVue();

  test("should have correct rendering", done => {
    const wrapper = mount(toastMessage, {
      localVue,
      attachToDocument: true
    });

    wrapper.vm.$nextTick(() => {
      expect(wrapper.isVueInstance()).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
      wrapper.destroy();
      done();
    });
  });
});
