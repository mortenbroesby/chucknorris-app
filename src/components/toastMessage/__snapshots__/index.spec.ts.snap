// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/toastMessage should have correct rendering 1`] = `
<div
  class="toastMessage"
>
  <div
    class="toastMessage--inner"
  >
    <div
      class="toastItem"
      style="display: none;"
    >
      <p
        class="toastItem--title"
      >
        Note:
      </p>
       
      <p
        class="toastItem--message"
      >
        
      </p>
    </div>
  </div>
</div>
`;
